cesium/internal/domain/writer.go

```diff
@@ -120,6 +120,12 @@ type Writer struct {
 	// fileKey represents the key of the file written to by the writer. One can convert it
 	// to a filename via the fileKeyToName function.
 	fileKey uint16
+	// fc is the file controller for the writer's FS.
+	fc *fileController
+	// fileSize is the writer's file's size
+	fileSize telem.Size
```

Comment:
Just curious as to why were tracking both `fileSize` and `lenWritten`

Reply:
Ah I see. We used to use the `Len()` method on `xio.TrackedWriteCloser`. Can we just get rid of this interface now? I think it's the only place we use it in the codebase. Can just use an `io.WriteCloser`.

Reply:
We still need the `Offset()` method – while we could bake that into the `domain/Writer` definition, I still like the writer having its own interface better, so I might instead just remove `Len()` and `Reset()` but keep offset. Which one would be the better approach?

Reply:
NVM – we still need both of those things: one to track how many bytes the domain writer has written, and the other to track how many bytes the individual trackedWriter has written – that needs to go into the pointer